File: .old-compat/src/main/java/androidx/preference/ListPreference.java
package androidx.preference;

import android.content.Context;

public class ListPreference extends Preference {
	private CharSequence[] entries, entryValues;
	
	public ListPreference(Context context) {}
	
	public CharSequence[] getEntries() {
		return entries;
	}
	
	public void setEntries(CharSequence[] entries) {
		this.entries = entries;
	}
	
	public int findIndexOfValue(String value) {
		throw new UnsupportedOperationException();
	}
	
	public CharSequence[] getEntryValues() {
		return entryValues;
	}
	
	public void setEntryValues(CharSequence[] entryValues) {
		this.entryValues = entryValues;
	}
	
	public void setValueIndex(int index) {
		throw new UnsupportedOperationException();
	}
	
	public String getValue() {
		throw new UnsupportedOperationException();
	}
	
	public void setValue(String value) {
		throw new UnsupportedOperationException();
	}
}

File: extension/loaders/android-compat/src/desktopMain/kotlin/androidx/preference/ListPreference.kt
package androidx.preference

import android.content.Context
import com.mrboomdev.awery.core.utils.PlatformSdk

@PlatformSdk
class ListPreference(context: Context): Preference(context) {
	private var entries: Array<CharSequence>? = null
	private var entryValues: Array<CharSequence>? = null

	@PlatformSdk
	fun getEntries(): Array<CharSequence>? {
		return entries
	}

	@PlatformSdk
	fun setEntries(entries: Array<CharSequence>?) {
		this.entries = entries
	}

	@PlatformSdk
	fun findIndexOfValue(value: String?): Int {
		if(value != null && entryValues != null) {
			for(i in entryValues!!.indices.reversed()) {
				// TODO: Uncomment once TextUtils implemented
//				if(TextUtils.equals(entryValues!![i].toString(), value)) {
//					return i
//				}
			}
		}
		return -1
	}

	@PlatformSdk
	fun getEntryValues(): Array<CharSequence>? {
		return entryValues
	}

	@PlatformSdk
	fun setEntryValues(entryValues: Array<CharSequence>?) {
		this.entryValues = entryValues
	}

	@PlatformSdk
	fun setValueIndex(index: Int) {
		throw RuntimeException("Stub!")
	}

	@PlatformSdk
	fun getValue(): String? {
		throw RuntimeException("Stub!")
	}

	@PlatformSdk
	fun setValue(value: String?) {
		throw RuntimeException("Stub!")
	}
}